pulser/register/base_register.py

```diff
@@ -52,7 +52,7 @@ class BaseRegister(ABC):
     """The abstract class for a register."""
 
     @abstractmethod
-    def __init__(self, qubits: Mapping[Any, ArrayLike]):
+    def __init__(self, qubits: Mapping[Any, ArrayLike], **kwargs: Any):
```

Comment:
How about taking `Optional[_LayoutInfo]` as input instead of `**kwargs: Any`, for better type safety and documentation?

Reply:
I though about that, but my intention is to hide this from the users. I want the way of associating a layout with a register to be only through the methods of `RegisterLayout` and its subclasses. To have a `layout_info` keyword argument in `Register` will give the impression that they should provide a layout when creating a `Register`, which is not the intended way.